hazelcast/src/main/java/com/hazelcast/nio/tcp/nonblocking/NonBlockingIOThread.java

```diff
@@ -56,43 +59,52 @@
     private final SwCounter selectorIOExceptionCount = newSwCounter();
     @Probe
     private final SwCounter completedTaskCount = newSwCounter();
+    // count number of times the selector was rebuilt (if selectWorkaround is enabled)
+    @Probe
+    private final SwCounter selectorRebuildCount = newSwCounter();
 
     private final ILogger logger;
 
-    private final Selector selector;
+    private volatile Selector selector;
```

Comment:
`NonBlockingIOThread.getSelector()` is invoked from `AbstractHandler.getSelectionKey` and `AbstractHandler.onFailure`. Apart from the former's usage in `NonBlockingSocketReader.init`, I think all other cases in which the `selector` is obtained come from the IO thread itself, so assuming the case of `init` invocation is irrelevant, `volatile` could be removed. Can you confirm?